src/components/ProfileForm/userform.js

```diff
@@ -0,0 +1,154 @@
+    userEmail: "",
+    dogs: []
+  };
+
+  async componentDidMount() {
+    const user = await Auth.currentAuthenticatedUser();
+    this.setState({ userEmail: user.attributes.email });
+    this.setState({ username: user.username });
+    console.log("UserForm current User", user);
+
+    this.getUser(user.attributes.sub);
+    this.getDogs(user.attributes.sub);
+  }
+
+  async getUser(userID) {
+    await API.graphql(graphqlOperation(getUser, { id: userID }))
+      .then(data =>
+        data.data.getUser != null
+          ? this.setState({
+              firstName: data.data.getUser.firstName,
+              lastName: data.data.getUser.lastName
+            })
+          : null
+      )
+      .catch(err => console.log(err));
+  }
+
+  async getDogs(userID) {
+    await API.graphql(graphqlOperation(getDogs, {id : userID}))
+      .then((data) => {
+        if (data.data.getDogs != null) {
+          // const updateDogs = data.data.getDogs
+          this.setState({
+            dogs : data.data.getDogs
+          })
+        }
+      })
+      .catch(err => console.log(err));
+  }
+
+  nextStep = () => {
+    const { step } = this.state;
+    this.setState({
+      step: 1
+    });
+  };
+
+  prevStep = () => {
+    const { step } = this.state;
+    this.setState({
+      step: step + 1
+    });
+  };
+
+  goDogs = () => {
+    const { step } = this.state;
+    this.setState({
+      step: 3
+    });
+  };
+
+  handleChange = input => e => {
+    this.setState({ [input]: e.target.value });
+  };
+
+  handleDogChange = (dogs) => {
+    this.setState({dogs : dogs})
+    console.log("change userform dogs state ", this.state.dogs)
+  }
+
+  showStep = () => {
+    const {
+      step,
+      // user
+      username,
+      firstName,
+      lastName,
+      userEmail,
+      dogs
+    } = this.state;
+
+    if (step === 1)
+      return (
+        console.log("current dogs in userform and sending to userprofileinfo ", dogs),
+        <div>
+          <Userprofileinfo
+            username={username}
+            firstName={firstName}
+            lastName={lastName}
+            userEmail={userEmail}
+            prevStep={this.prevStep}
+            goDogs={this.goDogs}
+            dogs={dogs}
+          />
+
+          {/* <Allinfo dogs={this.props.dogs} /> */}
+        </div>
+      );
+    if (step === 2)
+      return (
+        <div>
+          <UserInfo
+            handleChange={this.handleChange}
+            nextStep={this.nextStep}
+            username={username}
+            firstName={firstName}
+            lastName={lastName}
+            userEmail={userEmail}
+          />
+        </div>
+      );
+    if (step === 3)
+      return (
+        <div>
+          <Doginfodetails
+          dogs = {dogs}
+          handleDogChange={this.handleDogChange}
+          nextStep={this.nextStep}
+           />
+        </div>
+      );
+  };
+
+  render() {
+    const { step } = this.state;
+    return <>{this.showStep()}</>;
```

Comment:
doesn't this give a failure?

` return <>`

Reply:
@gaomigithub

Reply:
Nope. This almost equal to <div />; They do not affect the render effect.
@JohnLe9318 was used to type in this way. I changed almost of them to be <div /> but some of them just are neglected.
If anything like this is seen again, may just change them to <div />